apps/selvbetjening-api/src/main/kotlin/Application.kt

```diff
@@ -113,32 +112,51 @@ class ApplicationContext(
             }
         }
 
-    private fun kodeverkHttpClient() =
-        HttpClient(OkHttp) {
-            install(ContentNegotiation) {
-                jackson {
-                    configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-                    setSerializationInclusion(JsonInclude.Include.NON_NULL)
-                    registerModule(JavaTimeModule())
-                }
-            }
-        }
+    private fun kodeverkHttpClient(appConfig: Config) =
+        httpClientClientCredentials(
+            azureAppScope = "api://${appConfig.getString("kodeverk.client.id")}/.default"
+        )
+
 
     // OBS: Denne klienten kaller PDL med en systembruker.
     // Informasjon fra denne klienten kan inneholde informasjon sluttbruker ikke har rett til å se.
     private fun systemPdlHttpClient() =
-        HttpClient(OkHttp) {
-            install(ContentNegotiation) { jackson() }
-            install(Auth) {
-                clientCredential {
-                    config =
-                        System
-                            .getenv()
-                            .toMutableMap()
-                            .apply { put("AZURE_APP_OUTBOUND_SCOPE", requireNotNull(get("PDL_AZURE_SCOPE"))) }
-                }
-            }
+        httpClientClientCredentials(azureAppScope = System.getenv()["PDL_AZURE_SCOPE"]!!)
+}
+
+fun httpClientClientCredentials(
+    azureAppScope: String,
+) = HttpClient(OkHttp) {
+    install(ContentNegotiation) {
+        jackson {
+            configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            registerModule(JavaTimeModule())
         }
+    }
+    val env = System.getenv()
+
+    install(Auth) {
+        clientCredential {
+            config =
+                mapOf(
+                    AzureEnums.AZURE_APP_CLIENT_ID.name to env[AzureEnums.AZURE_APP_CLIENT_ID.name]!!,
+                    AzureEnums.AZURE_APP_JWK.name to env[AzureEnums.AZURE_APP_JWK.name]!!,
+                    AzureEnums.AZURE_APP_WELL_KNOWN_URL.name to env[AzureEnums.AZURE_APP_WELL_KNOWN_URL.name]!!,
+                    AzureEnums.AZURE_APP_OUTBOUND_SCOPE.name to azureAppScope,
+                )
+        }
+    }
+}
+
+enum class AzureEnums {
```

Comment:
Hvorfor ikke `Config`, `EnvVariables`, eller noe i den gaten i stedet for "enum"? Ikke veldig fan av å putte enum, class, e.l. i klassenavn

Reply:
https://github.com/navikt/pensjon-etterlatte/pull/1411/commits/6914d5e8ffd6aa7281d4f3b49bc8826f19be1d1c